### DISCUSSION TITLE
More controller input

### DISCUSSION THREAD

**Discussion by xlwnya** [vrchat-community/osc#49] [Ideas]
https://docs.vrchat.com/v2022.1.1/docs/osc-as-input-controller#supported-inputs
I feel some inputs are missing.

* Controllers & HMD tracking data inputs (It may be useful for emulate VR on desktop mode for example.)
* also for Trackers inputs. (emulate FBT)
* Gesture Toggle button.
* trigger emojis from OSC.
* Finger stretch inputs. (for emulating Valve Index controllers) also for outputs.
* GestureLeft, GestureRight, GestureLeftWeight, GestureRightWeight inputs. (control gestures from OSC)
* Viseme input (manually set Viseme from OSC)
* Finger spread inputs. (Quest hand tracking may have this tracking data)
* Valve index button/thumbstick touch inputs. (my controller's thumbstick touch sensor is broken…)。

**Comment by RicardoEscobar** [comment]
I'm trying to make facial expressions on VRChat from Python using `python-osc` library.
I already tried and foundout that `GestureLeft, GestureRight, GestureLeftWeight, GestureRightWeight inputs. (control gestures from OSC)` does not work outside unity.
Please check this post I did: [YouTube Video](https://youtu.be/jfHl4HEyDBU)
https://github.com/vrchat-community/osc/issues/42#issuecomment-1646451265
